### ISSUE TITLE
Multi steps form validation doesn't work on the second page and after

### ISSUE THREAD

**Issue by AnatoleLucet** [pablo-abc/felte#75]
Hey :wave: I'm having a small issue with multi steps forms.

I'm not sure if I'm doing it right, but with the following snippet, the first page does show the validation errors while the second doesn't:
```svelte
<script>
	import { createForms } from '@felte/multi-step';
	import { svelteReporter, ValidationMessage } from '@felte/reporter-svelte';
	import { validator } from '@felte/validator-yup';

	import * as yup from 'yup';

	const { pages, step, increaseStep } = createForms({
		initialStep: 0,
		pages: [
			{
				onSubmit: (values) => {
					console.log(values);
					increaseStep();
				},
				extend: [validator, svelteReporter],
				validateSchema: yup.object({
					name: yup.string().required()
				})
			},
			{
				onSubmit: (values) => {
					console.log(values);
				},
				extend: [validator, svelteReporter],
				validateSchema: yup.object({
					email: yup.string().required()
				})
			}
		]
	});

	$: form = pages[$step].form;
</script>

<form use:form>
	{#if $step === 0}
		<input type="text" name="name" placeholder="name" />
		<ValidationMessage for="name" let:messages={message}>
			{#if message}
				<span>
					{message}
				</span>
			{/if}
		</ValidationMessage>
	{:else}
		<input type="text" name="email" placeholder="email" />
		<ValidationMessage for="email" let:messages={message}>
			{#if message}
				<span>
					{message}
				</span>
			{/if}
		</ValidationMessage>
	{/if}

	<button type="submit">submit</button>
</form>
```

![](https://user-images.githubusercontent.com/35486736/142943305-d49d3b79-8329-43fa-9d52-d84b7f087769.gif)

Is this a bug or am I doing something wrong?

Thanks for creating this lib. Apart from this, my journey has been flawless using Felte

**Comment by pablo-abc** [comment]
Sorry about the delay. I'm aware there are some issues with the multi-step package as of now. Specially since I added support for `SolidJS` to the core package. It does need a major refactor that doesn't depend specifically on Svelte to adapt better to the future of Felte. As of now, the multi-step package itself is _really_ small, so worst case scenario you could see its source code and use it as an idea on how to handle multi-steps forms for yourself.

---

Basically we're creating one form per page (so one call to `createForm` per page needed), we have a "global" store that keeps track of the current page and we switch between components depending on the page. I'd recommend getting the `initialValues` for each form from the "global" store you create.

**Comment by pablo-abc** [comment]
After some consideration, and seeing how quickly this might change, I've updated the docs showcasing a possible solution to handle multi page forms (without using a package). For now, a custom solution like this seems to be way simpler than using a helper package. 

[Here's the documentation for it](https://felte.dev/docs/svelte/multi-page-forms), and there's a link to a [REPL example](https://svelte.dev/repl/8eb738732cf74edd86f680c53e6ba253?version=3.44.2) within it. Let me know if you find any issues with this and feel free to close this if this solves all of your issues!

**Comment by AnatoleLucet** [comment]
Thanks for the help. I'll try this in the coming days.

> I've updated the docs showcasing a possible solution to handle multi page forms (without using a package)

Yeah, I guess that makes sense considering how simple it is.
I'll close this for now and come back if I stumble upon anything that I think should be added to the doc. Thanks again.
